src/components/BrandConfigProvider/theme/generateColors.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  argbFromHex,
+  themeFromSourceColor,
+  Theme,
+} from '@importantimport/material-color-utilities';
+import Color from 'color';
+
+export function generateColors(color: string) {
+  const theme = themeFromSourceColor(argbFromHex(color));
+
+  const colorTheme = lightColorTheme(theme);
+
+  const mainColors = toRgbStrings(colorTheme);
+  const elevation = prepareElevations(colorTheme);
+  const surfaceColors = prepareSurfaceColors(colorTheme, theme.palettes);
+
+  return {
+    ...mainColors,
+    elevation,
+    ...surfaceColors,
+  };
+}
+
+type ColorTheme = ReturnType<typeof lightColorTheme>;
+
+const lightColorTheme = (theme: Theme) => {
+  return theme.schemes.light.toJSON();
+};
+
+const toRgbStrings = (theme: ColorTheme) => {
+  return Object.fromEntries(
+    Object.entries(theme).map(([key, value]) => [
+      key,
+      Color(value).rgb().string(),
+    ]),
+  );
+};
+
+const prepareElevations = (colorTheme: ColorTheme) => {
+  let elevations: Record<string, string> = {
+    level0: 'transparent',
+  };
+
+  const { primary, surface } = colorTheme;
+
+  const elevationLevels = [0.05, 0.08, 0.11, 0.12, 0.14];
+
+  elevationLevels.forEach((level, i) => {
+    elevations[`level${i + 1}`] = Color(surface)
+      .mix(Color(primary), Number(level))
+      .rgb()
+      .string();
+  });
+
+  return elevations;
+};
```

Comment:
Two parts of the theme that are required for RN Paper but are not a part of Material Design 3...

This is the first:

```TS
elevation: {
    level0: 'transparent',
    level1: 'rgb(238, 246, 246)',
    level2: 'rgb(230, 241, 242)',
    level3: 'rgb(223, 237, 238)',
    level4: 'rgb(220, 235, 237)',
    level5: 'rgb(215, 232, 234)',
}
```